core/frontend/src/components/app/SettingsMenu.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <v-container
+    class="d-flex justify-center"
+  >
+    <v-btn
+      class="mr-2"
+      icon
+      large
+      elevation="2"
+      @click="showDialog(true)"
+    >
+      <v-icon>mdi-cog</v-icon>
+    </v-btn>
+    <v-dialog
+      width="320"
+      :value="show_dialog"
+      @input="showDialog"
+    >
+      <v-card>
+        <v-card-title class="align-center">
+          Settings
+        </v-card-title>
+
+        <v-divider />
+
+        <v-container class="pa-8">
+          <v-row
+            justify="center"
+          >
+            <v-btn
+              @click="reset_settings"
+            >
+              <v-icon left>
+                mdi-trash-can
+              </v-icon>
+              Reset Settings
+            </v-btn>
+          </v-row>
+        </v-container>
+      </v-card>
+    </v-dialog>
+    <v-dialog
+      width="380"
+      :value="show_reset_dialog"
+      @input="show_reset_dialog = false"
+    >
+      <v-card>
+        <v-container class="pa-8">
+          Restart the system to finish the settings reset.
+        </v-container>
+      </v-card>
+    </v-dialog>
+  </v-container>
+</template>
+
+<script lang="ts">
+import axios from 'axios'
+import Vue from 'vue'
+import { getModule } from 'vuex-module-decorators'
+
+import NotificationStore from '@/store/notifications'
+import { commander_service } from '@/types/frontend_services'
+import { LiveNotification, NotificationLevel } from '@/types/notifications'
+
+const notification_store: NotificationStore = getModule(NotificationStore)
+
+const API_URL = '/commander/v1.0'
+
+export default Vue.extend({
+  name: 'PowerMenu',
+  components: {
+  },
+  data() {
+    return {
+      show_dialog: false,
+      show_reset_dialog: false,
+    }
+  },
+  computed: {
+  },
+  methods: {
+    async reset_settings(): Promise<void> {
+      await axios({
+        url: `${API_URL}/settings/reset`,
+        method: 'post',
+        params: {
+          i_know_what_i_am_doing: true,
```

Comment:
This variable seems to be a perfect fit for a confirmation dialog, although this should not block the PR.

Reply:
This variable is to avoid people using this REST API without confirming that they know what they are doing.

Reply:
Yes, I agree, this will be on the issue after this PR get merged.